webapp/src/client.js

```diff
@@ -15,29 +15,35 @@ export default class Client {
         return this.doPost(`${this.url}/settings`, meeting);
     }
 
-    doGet = async (url, body, headers = {}) => {
-        headers['X-Requested-With'] = 'XMLHttpRequest';
-        headers['X-Timezone-Offset'] = new Date().getTimezoneOffset();
-
-        const response = await request.
-            get(url).
-            set(headers).
-            accept('application/json');
-
-        return response.body;
+    doGet = async (url, headers = {}) => {
+        return this.doFetch(url, {headers});
     }
 
     doPost = async (url, body, headers = {}) => {
-        headers['X-Requested-With'] = 'XMLHttpRequest';
-        headers['X-Timezone-Offset'] = new Date().getTimezoneOffset();
+        return this.doFetch(url, {
+            method : 'POST',
+            body: JSON.stringify(body),
+            headers: {
+                ...headers, 
+                'Content-Type' : 'application/json',
+            }
+        });
+    }
 
-        const response = await request.
-            post(url).
-            send(body).
-            set(headers).
-            type('application/json').
-            accept('application/json');
+    doFetch = async (url, { method = 'GET', body = null, headers = {} }) => {
+        const options = Client4.getOptions({
+            method,
+            body,
+            headers: {
+                ...headers,
+                'X-Timezoone-Offset' : new Date().getTimezoneOffset(),
+                'Accept' : 'application/json'
+            },
+        });
 
-        return response.body;
+        const response = await fetch(url, options);
+
+        return response.json();
```

Comment:
Sorry about this, but I would like to request one more change regarding error handling. Just an additional improvement we can make as we're touching this here. Last request I promise :)

```suggestion
const response = await fetch(url, options);
 
let data;
if (response.ok) {
    data = await response.json();
    return {
        response,
        data,
    };
}

data = await response.text();
throw new ClientError(Client4.url, {
    message: data || '',
    status_code: response.status,
    url,
});
```

Reply:
I am mainly requesting the above because I usually knock out the CSRF and error handling fixes in the same PR. Let me know what you think

Reply:
> I am mainly requesting the above because I usually knock out the CSRF and error handling fixes in the same PR. Let me know what you think

Yes, it definitely makes sense from my point of view as well. What I don't understand however, is why if I catch this error and log it to the console I get the following: 
![image](https://user-images.githubusercontent.com/24561400/85944804-db324280-b939-11ea-8822-713517bf03a2.png)

I don't see the status code nor do I see the url. Instead I only get the message with some extra info, which I did not request.
Do you maybe know what I'm missing here?

Reply:
Could you paste the following here:

- The request/response payloads from dev tools
- Your code that is processing the error

You should be able to do something like this to log/handle the error that came from the server:

```js
try {
	await doFetch(url);
} catch (e) {
	handle(e.message);
}
```

Reply:
Interesting, didn't know that if you log the `e` to the console you just get it's message and **not the entire object**. Now that I've tried to get the individual properties, it works.

Reply:
> Should I now really handle it or did you mean just for debugging?

I was just giving an example to see if it could resolve the situation you describe above. As for if the error should be handled, it depends on the context, but either way that is out of scope for this PR.